dGame/dComponents/VendorComponent.cpp

```diff
@@ -129,12 +101,40 @@ void VendorComponent::SetupConstants() {
 	auto* vendorComponentTable = CDClientManager::Instance().GetTable<CDVendorComponentTable>();
 	std::vector<CDVendorComponent> vendorComps = vendorComponentTable->Query([=](CDVendorComponent entry) { return (entry.id == componentID); });
 	if (vendorComps.empty()) return;
-	m_BuyScalar = vendorComps[0].buyScalar;
-	m_SellScalar = vendorComps[0].sellScalar;
-	m_RefreshTimeSeconds = vendorComps[0].refreshTimeSeconds;
-	m_LootMatrixID = vendorComps[0].LootMatrixIndex;
+	auto vendorData = vendorComps.at(0);
+	m_BuyScalar = vendorData.buyScalar;
+	m_SellScalar = vendorData.sellScalar;
+	m_RefreshTimeSeconds = vendorData.refreshTimeSeconds;
+	m_LootMatrixID = vendorData.LootMatrixIndex;
 }
 
-bool VendorComponent::SellsItem(const LOT item) const {
-	return m_Inventory.find(item) != m_Inventory.end();
+
+void VendorComponent::MaxCustomVendorRefreshInventory(bool isCreation){
+	if(isCreation && m_Parent->GetLOT() == 9749 && Game::server->GetZoneID() == 1201) {
+		SetHasStandardCostItems(true);
+		m_Inventory.push_back(SoldItem(11909, 0)); // Top hat w frog
+		m_Inventory.push_back(SoldItem(7785, 0));  // Flash bulb
+		m_Inventory.push_back(SoldItem(12764, 0)); // Big fountain soda
+		m_Inventory.push_back(SoldItem(12241, 0)); // Hot cocoa (from fb)
+		return;
+	}
+
+	if (m_Parent->GetLOT() == 13569) {
+		auto randomCamera = GeneralUtils::GenerateRandomNumber<int32_t>(0, 2);
+
+		LOT camera = 0;
+		DluAssert(randomCamera >= 0 && randomCamera <= 2);
+		switch (randomCamera) {
+		case 0:
+			camera = 16253; // Grungagroid
+			break;
+		case 1:
+			camera = 16254; // Hipstabrick
+			break;
+		case 2:
+			camera = 16204; // Megabrixel snapshot
+			break;
+		}
+		m_Inventory.push_back(SoldItem(camera, 0));
+	}
```

Comment:
if we're going to make a special function for these items, lets put them into two separate functions since they operate on two different vendors.  Initially the name implied to me this was for the Max vendor but I didn't know until I read it that this also affects Mr. Ree.